### ISSUE TITLE
Error: Port seems to be closed: 80

### ISSUE THREAD

**Issue by miecio1212** [githubsaturn/captainduckduck#268]
```
docker run -p 80:80 -p 443:443 -p 3000:3000 -v /var/run/docker.sock:/var/run/docker.sock dockersaturn/captainduckduck
npm info it worked if it ends with ok
npm info using npm@5.4.2
npm info using node@v8.7.0
npm info lifecycle captain-duckduck@0.0.0~prestart: captain-duckduck@0.0.0
npm info lifecycle captain-duckduck@0.0.0~start: captain-duckduck@0.0.0

> captain-duckduck@0.0.0 start /usr/src/app
> node ./bin/www

Captain Starting ...
Installing Captain Service ...
 
 
 
 >>>> Checking System Compatibility <<<<
   Docker Version passed.
   Ubuntu detected.
   X86 CPU detected.
   Total RAM 16706 MB
 
Installation script of CaptainDuckDuck has recently changed.
Did your install script contain -p 80:80 -p 443:443 -p 3000:3000 ? If not, read "Getting Started" part of the docs!
If you are using the new script, and still seeing this message, read below!
 
 
Your firewall may have been blocking an in-use port: 80
A simple solution on Ubuntu systems is to run: ufw disable
See docs for more details on how to fix firewall issues
 
If you are an advanced user, and you want to bypass this check (NOT RECOMMENDED),
you can append the docker command with an addition flag: -e BY_PASS_PROXY_CHECK='TRUE'
 
 
Installation failed.
Error: Port seems to be closed: 80
    at Request._callback (/usr/src/app/src/utils/CaptainInstaller.js:172:24)
    at self.callback (/usr/src/app/node_modules/request/request.js:186:22)
    at emitOne (events.js:115:13)
    at Request.emit (events.js:210:7)
    at Request.onRequestError (/usr/src/app/node_modules/request/request.js:878:8)
    at emitOne (events.js:115:13)
    at ClientRequest.emit (events.js:210:7)
    at Socket.socketErrorListener (_http_client.js:385:9)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
npm info lifecycle captain-duckduck@0.0.0~poststart: captain-duckduck@0.0.0
npm info ok 
```
I tried this -> ufw disable

**Comment by githubsaturn** [comment]
Are you running this on a server with a public ip address or your local PC?

You cannot run CaptainDuckDuck on a local machine without some port forwarding. Captain is designed to be executed on servers with public IP address exposed.

**Comment by miecio1212** [comment]
on local PC but I've turned off firewall

**Comment by githubsaturn** [comment]
Running on a local PC is an advanced process. You need to do some port forwarding. Captain is a server management tool, and it requires public IP address to perform many actions. If you're an advanced user you can set up your system to run Captain locally, but keep in mind it's not a trivial process.
https://captainduckduck.com/docs/run-locally.html

**Comment by miecio1212** [comment]
`docker run -e BY_PASS_PROXY_CHECK='TRUE' -p 80:80 -p 443:443 -p 3000:3000 -v /var/run/docker.sock:/var/run/docker.sock dockersaturn/captainduckduck`
after I added this flag -e BY_PASS_PROXY_CHECK='TRUE' it started but I can't see localhost:3000 but when i CURL it returns html.

`docker service ps captain-captain --no-trunc` returns this
```
ID                          NAME                    IMAGE                                NODE                         DESIRED STATE       CURRENT STATE                      ERROR               PORTS
1vv4jpi0n78gds7ncdro2165n   captain-captain.1       dockersaturn/captainduckduck:0.7.2   mateusz-Lenovo-Y520-15IKBA   Ready               Preparing less than a second ago                       
24fouvc0irg01fjfzh0eknsgz    \_ captain-captain.1   dockersaturn/captainduckduck:0.7.2   mateusz-Lenovo-Y520-15IKBA   Shutdown            Running less than a second ago            
```
```
docker service logs captain-captain --follow
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | npm info it worked if it ends with ok
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | npm info using npm@5.4.2
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | npm info using node@v8.7.0
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | npm info lifecycle captain-duckduck@0.0.0~prestart: captain-duckduck@0.0.0
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | npm info lifecycle captain-duckduck@0.0.0~start: captain-duckduck@0.0.0
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | 
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | > captain-duckduck@0.0.0 start /usr/src/app
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | > node ./bin/www
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | 
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | Captain Starting ...
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:53.900 am    Emptying generated and temp folders.
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:53.906 am    Ensuring directories are available on host. Started.
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:53.910 am    Ensuring directories are available on host. Finished.
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:53.914 am    Network captain-overlay-network is already attached to service: captain-captain
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:53.917 am    Setting up NGINX conf file...
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:53.919 am    Locking NGINX configuration reloading...
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:53.922 am    SUCCESS: UNLocking NGINX configuration reloading...
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:53.923 am    Captain Nginx is already running.. 
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:53.925 am    Updating NGINX service...
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:53.933 am    Waiting for 5 seconds for nginx reload to take into effect
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:58.940 am    NGINX is fully set up and working...
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:58.958 am    captain-captain (service) has already been connected to secret: captain-salt
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:58.960 am    There is no Docker Registry, neither local nor remote.
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:58.968 am    No Captain Certbot service is running. Creating one...
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:50:58.975 am    Waiting for 5 seconds for Certbot to start up
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:51:03.977 am    Updating Certbot service...
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:51:03.996 am    **** Captain is initialized and ready to serve you! ****
captain-captain.1.9crwlugbizo8@mateusz-Lenovo-Y520-15IKBA    | September 5th 2018, 8:52:04.019 am    executeCommand Container: e3d7363ff3b1a4748549715a8b49647898967d79cc80cc1e64343d6e0af6214f
```

**Comment by githubsaturn** [comment]
The reason is that your domain name is not resolved to the correct IP address inside the container. This could be because your Docker Swarm is using another network interface that docker network. Running Captain on a local machine is only for Captain's development purposes, it can get quite complicated and buggy. Captain is solely designed to be executed on a server.

**Comment by AkgunFatih** [comment]
im using vultr vps and have the same problem

**Comment by AkgunFatih** [comment]
@githubsaturn 

I'm using [DockSTARTer](https://github.com/GhostWriters/DockSTARTer) with letsencrypt enabled

**Comment by githubsaturn** [comment]
DockSTARTer is a 3rd party application untested and unsupported. It may or may not cause problems.

CapRover is tested on Ubuntu with Docker. In fact, I justed tested CapRover specifically on Vultr and it just worked fine. I chose Docker on Ubuntu from Applications on Vultr control panel.

PS: Make sure you DO NOT enable IPv6 as well.


 
![screenshot from 2019-01-24 18-02-31](https://user-images.githubusercontent.com/10582082/51720403-4881cb80-2002-11e9-8c3e-e8e4522bb03b.png)
